.github/workflows/integration-test.yml

```diff
@@ -28,6 +28,23 @@ env:
   TARGET_PYTHON_VERSION: "3.9"
 
 jobs:
+  docs-check:
+    # This job is used to check the documentation build with strict mode enabled
+    name: Docs check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+      - name: Setup python, and load cache
+        uses: ./.github/actions/setup-env
+        with:
+          python-version: ${{ env.TARGET_PYTHON_VERSION }}
+          cache-pre-commit: false
+          cache-venv: true
+          setup-poetry: true
+          install-deps: true
+      - name: Build documentation
+        run: poetry run mkdocs build --strict
```

Comment:
Is there a way to build this check into a pre-GH-Actions check so developers can avoid surprising results from possible failures here? Mostly I wonder here if there's a pre-commit check which could be included that runs a check that things are prepared in advance.